unicodetools/org/unicode/text/utility/Settings.java

```diff
@@ -70,11 +76,58 @@ private static final String getRequiredPathAndFix(String key) {
         public static final String UNICODETOOLS_DIR = UNICODETOOLS_REPO_DIR + "unicodetools/";
         public static final String UNICODEJSPS_DIR = UNICODETOOLS_REPO_DIR + "UnicodeJsps/";
         public static final String DATA_DIR = UNICODETOOLS_DIR + "data/";
+        public static final Path DATA_FILE = Paths.get(DATA_DIR);
         public static final String UCD_DIR = DATA_DIR + "ucd/";
         // TODO: IDN_DIR is used, but there is no .../data/IDN/ folder. Should this be .../data/idna/ ?
         public static final String IDN_DIR = DATA_DIR + "IDN/";
         // TODO: DICT_DIR is used, but there is no .../data/dict/ folder. ??
         public static final String DICT_DIR = DATA_DIR + "dict/";
+        
+        public enum DataDir {
+        	security,
+        	ucd,
+        	idna,
+        	emoji;
+        	
+        	/**
+        	 * This dir as a Path
+        	 * @return
+        	 */
+        	public Path asPath() {
+        		return  DATA_FILE.resolve(name());
+        	}
+        	/**
+        	 * This dir as a Path to the version subdir
+        	 * @param forVersion
+        	 * @return
+        	 */
+        	public Path asPath(VersionInfo forVersion) {
+        		String versionString = versionToString(forVersion);
+				if (this == ucd) {
+        			// For some reason, these have -Update
+        			return asPath().resolve(versionString + "-Update");
+        		} else {
+        			return asPath().resolve(versionString);
+        		}
+        	}
+        	/**
+        	 * Map a version number to a string.
+        	 * @param version
+        	 * @return
+        	 */
+        	public String versionToString(VersionInfo version) {
```

Comment:
There is a method on VersionInfo that produces a string with specified fields. Should be used instead of string building. Could be fixed later if you want.

Reply:
I saw that but it is ICU `@Internal` and has its own behavior.

Reply:
> We use it in a bunch of places, and frankly there isn't a good reason for it not to be public. Mark

But this isn't a blocker at all!!